llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1383,7 +1385,13 @@ InstructionCost RISCVTTIImpl::getCmpSelInstrCost(unsigned Opcode, Type *ValTy,
            getRISCVInstructionCost(RISCV::VMSLT_VV, LT.second, CostKind);
   }
 
-  if ((Opcode == Instruction::FCmp) && ValTy->isVectorTy()) {
+  if ((Opcode == Instruction::FCmp) && ValTy->isVectorTy() &&
+      CmpInst::isFPPredicate(VecPred)) {
+
+    // Use VMXOR_MM and VMXNOR_MM to generate all true/false mask
+    if ((VecPred == CmpInst::FCMP_FALSE) || (VecPred == CmpInst::FCMP_TRUE))
+      return getRISCVInstructionCost(RISCV::VMXOR_MM, LT.second, CostKind);
```

Comment:
Isn't LT here based on ValTy not CondTy? Or does getRISCVInstructionCost not care about the type for VMXOR?

Reply:
Since getLMUL(MVT) interprets `<vscale x N x i1>` as `<vscale x N x i8>`, getRISCVInstructionCost is unable to derive the LMUL information from the type for mask operations.

Reply:
Hopefully VMXOR_MM doesn't use the LMUL? Since a mask always fits in VLEN or fewer bits, they shouldn't be affected by LMUL.

Reply:
Yes, the cost for mask operations is hardcoded to be 1.